### ISSUE TITLE
Refactor APIRef/groupData macros for event reference changes

### ISSUE THREAD

**Issue by Elchi3** [mdn/sprints#906]
This is a work item for #685.

The APIRef/groupData macros need updating when we change the event reference.

See https://github.com/mdn/kumascript/blob/master/macros/APIRef.ejs
This macro is used on like 11,000 pages in all locales and has a function called buildEventList which gets events of the API from groupData and then links to them in the now to be removed Web/Events/ tree. 

We need to either rip this out or update it so that it can link to the events that will be in the Web/API/ doc tree.

It looks like same needs to be done for https://github.com/mdn/kumascript/blob/master/macros/DefaultAPISidebar.ejs and https://github.com/mdn/kumascript/blob/master/macros/InterfaceOverview.ejs but I'm not sure these two macros should be alive.

**Comment by wbamberg** [comment]
I think we should have tests for APIRef before doing this. We should have tests for APIRef anyway.

Note that @ExE-Boss is working on changes to APIRef.ejs, DefaultAPISidebar.ejs, and EventRef.ejs over in https://github.com/mdn/kumascript/pull/1041 (as part of moving the JSON macros into mdn/data), and I asked for tests as a precondition of landing that. From what I can tell the PR now has tests for DefaultAPISidebar.ejs, and EventRef.ejs but not APIRef.ejs.

This work also connects with https://github.com/mdn/sprints/issues/940, in which someone, I believe @escattone , is going to add tests for high-priority macros.

So I'd suggest we do something like: split out the test portion of https://github.com/mdn/kumascript/pull/1041 into a separate PR and add tests for APIRef.ejs. I think @escattone might want to do that, but if he doesn't @ExE-Boss might want to, and if they don't, I will :).

This would unblock a bunch of work, including this and the rest of https://github.com/mdn/kumascript/pull/1041.

We will I think remove EventRef.ejs as part of this work, so I don't think we need to keep those tests from https://github.com/mdn/kumascript/pull/1041.

As for InterfaceOverview.ejs, I think we should remove it. I think @a2sheppy might not be happy about that, but it's only used on 8 pages on all locales, so isn't really earning its keep, and I don't think adding more complex macros is the direction we want to go in future. If we do decide we want it in future, well, we can add it back then.

**Comment by ExE-Boss** [comment]
I hadn’t yet really gotten around to&nbsp;doing tests for&nbsp;`{{APIRef}}` because it’s&nbsp;one of&nbsp;the&nbsp;most complex sidebar macros in&nbsp;KumaScript at&nbsp;the&nbsp;time of&nbsp;this&nbsp;writing.

Anyway, I’ll split the&nbsp;currently written tests portion into a&nbsp;separate&nbsp;PR.

**Comment by wbamberg** [comment]
Apart from the testing piece, the updates to APIRef.ejs would I guess be to make it treat events in more or less the same way it currently treats constructors, methods, and properties: get the pages under the current page that are tagged "Event" and add them to their own section of the sidebar. If we want to use a `_event` suffix for event pages we would have to remove that (for this reason among others I don't really like the suffix idea, but it might be the best option available).

This is somewhat like the way the WebExtension docs deal with events: https://github.com/mdn/kumascript/blob/master/macros/WebExtAPISidebar.ejs#L156-L158.

**Comment by Elchi3** [comment]
Thanks for your feedback, Will!

Removing EventRef and InterfaceOverview sounds good to me.
Then first testing APIRef before updating it for the new events location also makes sense.
I'm also happy to dig into related open KumaScript PRs by @ExE-Boss or others as part of this work. I haven't estimated this issue yet, but I would like to take some serious amount of time to get it right.

> The updates to APIRef.ejs would I guess be to make it treat events in more or less the same way it currently treats constructors, methods, and properties: get the pages under the current page that are tagged "Event" and add them to their own section of the sidebar. If we want to use a `_event` suffix for event pages we would have to remove that (for this reason among others I don't really like the suffix idea, but it might be the best option available).

This would have been my plan as well! Thanks for writing it out. I think we've agreed on the suffix when we made the plan, but this week would be the time to argue about it. I have already moved pages for the first API and used the suffix there.

**Comment by a2sheppy** [comment]
Please don't remove InterfaceOverview. It's a macro I've been working on actively (despite not having committed updates recently), in preparation for changing a good bit of content to make use of it. If you remove it, I'm just going to have to put it back again, and that's just wasting people's time. :)

**Comment by a2sheppy** [comment]
And as @ExE-Boss says, the broken output has a patch but I can't get it committed because I can't get a review without tests, and when I tried to write tests, they wouldn't work. So I need to make another pass at trying to write tests for it.

**Comment by chrisdavidmills** [comment]
Yeah, please leave sheppy's work intact until he has a chance to fix it up and prove its worth ;-)

**Comment by wbamberg** [comment]
>  leave sheppy's work intact

The problem is that the event work will break InterfaceOverview, and InterfaceOverview doesn't have tests. So we have various options:

1. update it and write tests (and fix https://github.com/mdn/kumascript/pull/691 as well I suppose) 
2. update it and don't write tests, but then how do we know it's updated correctly?
3. leave it broken (not an option)
4. remove it

I guess the plan would be for option (1) then?

My view is that this is a disproportionate amount of work for a macro that's only used on 8 pages, when it would be easy to take it out now and reinstate it when it has been updated, if that's the direction we want to go in. Honestly I'm also a bit reluctant to extend the reach of KS macros at this point.

But if this is the decision, then OK.

**Comment by wbamberg** [comment]
Hmm, as far as I can tell, while InterfaceOverview [does load GroupData](https://github.com/mdn/kumascript/blob/master/macros/InterfaceOverview.ejs#L41), and [does access the events property](https://github.com/mdn/kumascript/blob/master/macros/InterfaceOverview.ejs#L81), it doesn't do anything with it - the `events` variable is unused after initialisation.

So it looks like we could just delete those lines and nothing would change, and we would remove the dependency on GroupData entirely.

**Comment by ExE-Boss** [comment]
Apparently the&nbsp;goal seems to&nbsp;have been to&nbsp;generate an **Events** section like the&nbsp;one present on&nbsp;some&nbsp;API interface&nbsp;docs, but it&nbsp;hasn’t yet been implemented.

**Comment by chrisdavidmills** [comment]
>Hmm, as far as I can tell, while InterfaceOverview [does load GroupData](https://github.com/mdn/kumascript/blob/master/macros/InterfaceOverview.ejs#L41), and [does access the events property](https://github.com/mdn/kumascript/blob/master/macros/InterfaceOverview.ejs#L81), it doesn't do anything with it - the `events` variable is unused after initialisation.

>So it looks like we could just delete those lines and nothing would change, and we would remove the dependency on GroupData entirely.

Sorry, I meant to reply to thius last night, but then ran out of time. So if this is the case, I'd support this as a bandaid for now to allow us to continue with the events work.

**Comment by Elchi3** [comment]
This would be my vote as well. If we're not allowed to remove the macro, I would like to at least unblock us by removing the event part of the implementation there. Thanks Chris.

**Comment by ExE-Boss** [comment]
I&nbsp;am&nbsp;addressing the&nbsp;`{{InterfaceOverview}}` issues in&nbsp;https://github.com/mdn/kumascript/pull/1073, which&nbsp;supersedes&nbsp;https://github.com/mdn/kumascript/pull/691.

**Comment by wbamberg** [comment]
I've split this user story up into:

https://github.com/mdn/sprints/issues/1313
https://github.com/mdn/sprints/issues/1314
https://github.com/mdn/sprints/issues/1315
